File: runtimes/python_template/bot_template.py
"""Reference webhook bot template (Python).

Drop your own logic into `handle_update`. The platform sets:
  BOT_TOKEN     — your Telegram token
  PORT          — local TCP port to bind on (loopback)
  WEBHOOK_PATH  — path the platform forwards updates to
"""
from __future__ import annotations

import json
import os

import httpx
import uvicorn
from fastapi import FastAPI, Request

TOKEN = os.environ.get("BOT_TOKEN", "")
PORT = int(os.environ.get("PORT", "0") or 0) or 5000
PATH = os.environ.get("WEBHOOK_PATH", "/webhook")
TG = f"https://api.telegram.org/bot{TOKEN}"

app = FastAPI()


async def handle_update(update: dict, http: httpx.AsyncClient) -> None:
    msg = update.get("message")
    if msg and msg.get("text"):
        chat_id = msg["chat"]["id"]
        await http.post(f"{TG}/sendMessage", data={
            "chat_id": chat_id,
            "text": "👋 أهلاً! أنا بوتك المُستضاف على TikZoom.",
        })


@app.post("/webhook")
async def webhook(request: Request) -> dict:
    update = await request.json()
    async with httpx.AsyncClient(timeout=10.0) as http:
        await handle_update(update, http)
    return {"ok": True}


if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=PORT)

File: runtimes/php_template/bot_template.php
<?php
/**
 * Reference webhook bot template (PHP).
 *
 * Run via:  php -S 127.0.0.1:$PORT -t .  (the platform handles the runner)
 *
 * Environment variables provided by the platform:
 *   BOT_TOKEN     — your Telegram token
 *   PORT          — local TCP port to bind on
 *   WEBHOOK_PATH  — path the platform forwards updates to (default /webhook)
 */

$token = getenv('BOT_TOKEN') ?: '';
$path  = getenv('WEBHOOK_PATH') ?: '/webhook';

$reqPath = parse_url($_SERVER['REQUEST_URI'] ?? '/', PHP_URL_PATH);

if ($reqPath !== $path) {
    http_response_code(404);
    echo json_encode(['ok' => false, 'error' => 'not found']);
    return;
}

$body = file_get_contents('php://input');
$update = json_decode($body, true);

if (isset($update['message']['text'])) {
    $chatId = $update['message']['chat']['id'];
    $url = "https://api.telegram.org/bot{$token}/sendMessage";
    $payload = http_build_query([
        'chat_id' => $chatId,
        'text'    => '👋 أهلاً من بوت PHP المُستضاف على TikZoom.',
    ]);
    $ctx = stream_context_create([
        'http' => ['method' => 'POST', 'header' => 'Content-Type: application/x-www-form-urlencoded',
                   'content' => $payload, 'timeout' => 10]
    ]);
    @file_get_contents($url, false, $ctx);
}

header('Content-Type: application/json');
echo json_encode(['ok' => true]);
